cardinal/ecs/component/component.go

```diff
@@ -14,11 +14,15 @@ type (
 		ID() TypeID
 		// New returns the marshaled bytes of the default value for the component struct.
 		New() ([]byte, error)
-		// Name returns the name of the component.
-		Name() string
 
-		Decode([]byte) (any, error)
 		Encode(any) ([]byte, error)
+		Decode([]byte) (any, error)
+		INameable
+	}
+
+	INameable interface {
```

Comment:
nit: i dont think INameable communicates that this is something related to components.

how about `IAbstractComponent`? Open to alternatives.

Reply:
Sure that works.

Reply:
Done. We can always change this later as well.

Reply:
I don't think we should go with IAbstractComponent. too vague imo. I would also like to get away from prefixing all our interfaces with `I`. 

in some places it makes sense, i.e. differenciating the `ComponentType` generic struct and the interface equivalent, but i don't think using it as a first resort is a pattern we should adopt

can we think of some others here? `type NamedComponent interface`?

Reply:
idc too much for this one, but I see scotts perspective. The user is creating a component. His ONLY interaction with anything called component is through this interface. IComponent and ComponentType becomes private and the user doesn't use or interact with it anymore.  

Therefore, from the user perspective it makes more sense to call it a Component or something along those lines, from the dev perspective it makes sense to call it Nameable because that interface can literally be used with anything that has a name method NOT just components. 

In fact, because this is user facing it might be better to have this Interface take up the IComponent mantle and have all the other types be called something like IPrivateComponent, or PrivateComponentType as the user should no longer be interacting with these primitives. 

Additionally, prefixing interfaces with I is both a common pattern in other languages and a useful/practical one as Structs and Interfaces are used in the same places interchangeably. What is your reasoning for why you don't want to prefix it with an I?

Reply:
Agree with everything there 

> Additionally, prefixing interfaces with I is both a common pattern in other languages and a useful/practical one as Structs and Interfaces are used in the same places interchangeably. What is your reasoning for why you don't want to prefix it with an I?

It's not really common in Go. I haven't seen any interfaces provided by Go's std libs that prefix with `I`. https://sweetohm.net/article/go-interfaces.en.html

would prefer to keep this as Go-aligned as possible. If there are compelling reasons to delineate, I'm OK with it, but prefer not to default to `I` prefixes.

Reply:
Eh. I disagree with this convention. But of course, you're right, we should follow the convention if it exists because that's what users expect.